### PR TITLE
Add basic Telegram Functionality

### PR DESCRIPTION
**Pull request by SgtBatten**
Improvements for #31

### CODE REVIEW

File: Frigate Camera Notifications/Beta [hunk 1]
```diff
@@ -887,6 +889,18 @@ action:
...
+                      - conditions: "{{ telegram }}"
+                        sequence:
+                          - service: "notify.{{ telegram_target }}"
+                            data:
+                              title: "{{title}}"
+                              message: "{{message}}"
+                              data:
+                                message_tag: "{{ id }}"
+                                photo:
+                                  - url: "{{base_url}}/api/frigate{{client_id}}/notifications/{{id}}/snapshot.jpg"
```
Comment (Chrisgozd) [inline Frigate Camera Notifications/Beta:901]:
```suggestion
                                  - url: "{{base_url}}/api/frigate{{client_id}}/notifications/{{id}}/{{attribute}}.jpg"
```

**Comment by wouterrutgers** [comment]
Hi there! Any progress on this PR?

Currently, I'm experiencing delays of 5-30 minutes with my home assistant notifications due to a bug in the system (https://github.com/home-assistant/android/issues/3396).

As a temporary solution, I have started using Telegram to receive notifications, but using the stable branch only text notifications are sent to Telegram.

Therefore I've tried using this branch, but this branch seems to be out of date and any area checks in the filters are skipped. This means that I'm receiving notifications for people who are not in the area I have defined.

I'm willing to help test the telegram integration, but I think it would be best to first update this branch to be up to date with the main branch. 

Thanks!

**Comment by borgqueenx** [comment]
hi, I'd love to help test and debug this, and thanks a lot for wanting to add telegram functionality. 

You asked for a working blueprint, the following blueprint is working for me, but has some bugs and limited functionality.
It has been editted by me to fit my needs a little bit, but you probably get a idea of it.

The bugs are the delays that are needed. if i do not use those delays, the video is send as soon as a object enters a zone, and therefore i do not get a full video event. and the video link that is send, is often not yet ready to be viewed and need to click it a little later for it to be working. 
Also often i am not receiving any video link, just a snapshot, but might have traced that down to a timeout of the telegram bot, that i now changed to continue if it timeouts, did not test that yet in below blueprint: 

`blueprint:
  name: Frigate - Telegram Notification
  description: Create automations to receive Snapshots and Clips from Frigate
  domain: automation
  input:
    camera:
      name: Frigate Camera
      description:
        The name of the camera as defined in your frigate configuration
        (/conf.yml).
    target_chat:
      name: Target
      description:
        "The chat_id to be used by the Telegram bot. !secret chat_id is
        not allowed on Blueprint, you will need the chat_id code.

        "
    notification:
      name: Notification
      description:
        'Select "true" to disable notification, lave "false" to receive
        notification. '
      selector:
        select:
          options:
            - "true"
            - "false"
          multiple: false
          custom_value: false
          sort: false
      default: "false"
    base_url:
      name: (Optional) Base URL
      description:
        The external url for your Home Assistant instance. This will default
        to a relative URL and will open the clips in the app instead of the browser,
        which may cause issues on some devices.
      default: ""
    zone_filter:
      name: (Optional) Zone Filter
      description: Only notify if object has entered a defined zone.
      default: false
      selector:
        boolean: {}
    zones:
      name: (Optional) Trigger Zones
      description: A list (-) of zones you wish to recieve notifications for.
      default: []
      selector:
        object: {}
    labels:
      name: (Optional) Trigger Objects
      description: A list (-) of objects you wish to recieve notifications for.
      default: []
      selector:
        object: {}
    presence_filter:
      name: (Optional) Presence Filter
      description: Only notify if selected presence entity is not "home".
      default: ""
      selector:
        entity:
          multiple: false
  source_url: https://gist.github.com/NdR91/5486a1e55101e062c48545395b7dd9a3#file-frigate_telegram_notification-yaml
mode: single
max_exceeded: silent
trigger:
  platform: mqtt
  topic: frigate/events
  payload: !input camera
  value_template: "{{ value_json['after']['camera'] }}"
variables:
  id: "{{ trigger.payload_json['after']['id'] }}"
  camera: "{{ trigger.payload_json['after']['camera'] }}"
  camera_name: "{{ camera | replace('_', ' ') | title }}"
  target_chat: !input target_chat
  object: "{{ trigger.payload_json['after']['label'] }}"
  label: "{{ object | title }}"
  entered_zones: "{{ trigger.payload_json['after']['entered_zones'] }}"
  type: "{{ trigger.payload_json['type'] }}"
  base_url: !input base_url
  zone_only: !input zone_filter
  input_zones: !input zones
  zones: "{{ input_zones | list }}"
  input_labels: !input labels
  labels: "{{ input_labels | list }}"
  presence_entity: !input presence_filter
  notification: !input notification
condition:
  - "{{ type != 'end' }}"
  - "{{ not zone_only or entered_zones|length > 0 }}"
  - "{{ not zones|length or zones|select('in', entered_zones)|list|length > 0 }}"
  - "{{ not labels|length or object in labels }}"
  - "{{ not presence_entity or not is_state(presence_entity, 'home') }}"
action:
  - service: telegram_bot.send_photo
    data:
      target: "{{ target_chat }}"
      disable_notification: "{{ notification }}"
      caption: "{{ camera_name }} Triggered! Watch Live: http://192.168.1.184:1984/stream.html?src={{ camera }}"
      url: "{{base_url}}/api/frigate/notifications/{{id}}/snapshot.jpg"
  - repeat:
      sequence:
        - wait_for_trigger:
            - platform: mqtt
              topic: frigate/events
              payload: "{{ id }}"
              value_template: "{{ value_json['after']['id'] }}"
          timeout:
            minutes: 3
          continue_on_timeout: true
        - condition: template
          value_template: "{{ wait.trigger.payload_json['type'] == 'end' }}"
        - delay: "00:00:06"
        - service: telegram_bot.send_message
          data:
            target: "{{ target_chat }}"
            disable_notification: "{{ notification }}"
            message: "{{ camera_name }} event link: {{base_url}}/api/frigate/notifications/{{id}}/{{camera}}/clip.mp4 ID={{id}}"
        - delay: "00:00:03"
        - service: telegram_bot.send_video
          data:
            target: "{{ target_chat }}"
            disable_notification: "{{ true }}"
            url: "{{base_url}}/api/frigate/notifications/{{id}}/{{camera}}/clip.mp4"
        - delay: "00:00:01"
        - service: telegram_bot.send_message
          data:
            target: "{{ target_chat }}"
            disable_notification: "{{ true }}"
            message: "------------ Event End ------------"
      until: "{{ wait.trigger.payload_json['type'] == 'end' }}"
`

Let me know how to get the newest beta blueprint so i can test it.

**Comment by malkstar** [comment]
Thanks for this, it works for my usecase (just need the snapshot, rather than video). I have been using manual event catching until now, and living with duplicate events spamming my telegram. 

I don't actually have clips setup right now, but I will set them up to test the video aspect of the PR.

**Comment by Rooster212** [comment]
Hello, I stumbled on this PR after wanting to setup Telegram notifications, I'm happy to test and do some feedback.

* During setup, I had to enter a device and a chat ID - otherwise I got a `Malformed device 'false'` message I think.
* I don't appear to be able to get a video with any combination of settings - however thumbnail and snapshot settings both work for me. Perhaps this is as per the above comment, with clips - not 100% sure if that is setup in my Frigate instance.

**Comment by SgtBatten** [comment]
Thanks

> * During setup, I had to enter a device and a chat ID - otherwise I got a `Malformed device 'false'` message I think.
HA update 2023.11 caused this on all versions. No functionality impact, just have to have a device even if sending to a group. group overrides the device if used.

**Comment by bozzio83** [comment]
I cant get any notification when using the group feature. I have a telegram and signal notify in the group.  When i test the service i get both notifications. But I can't get this blueprint to send anything?

**Comment by owaismai** [comment]
> I cant get any notification when using the group feature. I have a telegram and signal notify in the group. When i test the service i get both notifications. But I can't get this blueprint to send anything?

I have telegram notifications working. It does not seem to work though on the latest versions of Home Assistant. 
Its working on ha os 10.5 and ha core 2023.9.3
I have latest version of frigate addon and Frigate integration v5.2.0
Using Frigate Notifications (0.12.0.1h) (the latest beta version of @SgtBatten script as of now)

Its working on the above. When I upgrade ha core and ha os it no longer works. 

I really wish it can send video to telegram.

**Comment by SgtBatten** [comment]
> (0.12.0.1h)

This isn't the latest, but I've also not done anything for telegram in a long time as I need more assistance

**Comment by owaismai** [comment]
How can I assist?

**Comment by owaismai** [comment]
> > (0.12.0.1h)
> 
> This isn't the latest, but I've also not done anything for telegram in a long time as I need more assistance

I dont understand, isnt this the latest beta that supports telegram?
https://github.com/SgtBatten/HA_blueprints/blob/telegram/Frigate%20Camera%20Notifications/Beta

**Comment by SgtBatten** [comment]
As far as im aware telegram works in the main branch. I don't recall what the differences are between the two for telegram specifically.

**Comment by SgtBatten** [comment]
closed by #315

**Pull request closed without merging**